src/main/java/com/suse/saltstack/netapi/event/EventStream.java

```diff
@@ -183,30 +199,63 @@ public void onOpen(Session session, EndpointConfig config) throws IOException {
      *
-     * @param message The message received on this WebSocket
+     * @param partialMessage partial message received on this websocket
+     * @param last indicate the last part of a message
+     * @throws MessageTooBigException in case the message is longer than maxMessageLength
      */
     @OnMessage
-    public void onMessage(String message) {
-        if (message != null && !message.equals("server received message")) {
-            // Salt API adds a "data: " prefix that we need to ignore
-            Event event = JsonParser.EVENTS.parse(message.substring(6));
-            synchronized (listeners) {
-                listeners.stream().forEach(l -> l.notify(event));
+    public void onMessage(String partialMessage, boolean last)
+            throws MessageTooBigException {
+        if (last) {
+            String message;
+            if (partialMessageBuffer.length() == 0) {
+                message = partialMessage;
+            } else {
+                partialMessageBuffer.append(partialMessage);
+                message = partialMessageBuffer.toString();
+            }
+            if (maxMessageLength >= 0 && message.length() > maxMessageLength) {
+                throw new MessageTooBigException(maxMessageLength);
+            }
+
+            // Notify all registered listeners
+            if (!message.equals("server received message")) {
+                // Salt API adds a "data: " prefix that we need to ignore
+                Event event = JsonParser.EVENTS.parse(message.substring(6));
+                synchronized (listeners) {
+                    listeners.stream().forEach(l -> l.notify(event));
+                }
+            }
+
+            // Reset the size to the defaultBufferSize and empty the buffer
+            partialMessageBuffer.setLength(defaultBufferSize);
+            partialMessageBuffer.trimToSize();
+            partialMessageBuffer.setLength(0);
+        } else {
+            partialMessageBuffer.append(partialMessage);
+            if (maxMessageLength >= 0 && partialMessageBuffer.length() > maxMessageLength) {
+                throw new MessageTooBigException(maxMessageLength);
             }
         }
     }
 
     /**
-     * On error, close all objects of this class.
+     * On error, convert {@link Throwable} into {@link CloseReason} and close the session.
      *
      * @param t The Throwable object received on the current error.
+     * @throws IOException in case of an error when closing the session
      */
     @OnError
-    public void onError(Throwable t) {
-        this.close();
-        t.printStackTrace();
+    public void onError(Throwable t) throws IOException {
+        CloseReason closeReason;
+        if (t instanceof MessageTooBigException) {
+            closeReason = new CloseReason(CloseCodes.TOO_BIG, t.getMessage());
+        } else {
+            closeReason = new CloseReason(CloseCodes.CLOSED_ABNORMALLY, t.getMessage());
+        }
+        close(closeReason);
```

Comment:
Done (and renamed `t` to `throwable`) with 746bd688a493797de86e9dae1216c2d05de8d003.